pkgs/development/libraries/libredwg/default.nix

```
@@ -0,0 +1,42 @@
+{ lib, stdenv, fetchFromGitHub, autoreconfHook, pkgconfig, texinfo, pcre2
+, enablePython ? false, python, swig, libxml2, ncurses
+}:
+let
+  isPython3 = enablePython && python.pythonAtLeast "3";
+in
+stdenv.mkDerivation rec {
+  pname = "libredwg";
+  version = "0.10.1";
+
+  src = fetchFromGitHub {
+    owner = "LibreDWG";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256:1zd721z2nriw1jlrh4y1fj59b0dnymhd4kwp8rqw16bs84gda37n";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkgconfig texinfo ];
+
+  buildInputs = [ pcre2 ]
+    ++ lib.optionals enablePython [ python swig libxml2 libxml2.dev ]
+    # configurePhase fails with python 3 when ncurses is missing
+    ++ lib.optional isPython3 ncurses
+  ;
+
+  # prevent python tests from running when not building with python
+  configureFlags = lib.optional (!enablePython) "--disable-python";
+
+  doCheck = true;
+
+  checkPhase = ''
+    make check
+  '';
+
+  meta = with lib; {
+    description = "LibreDWG - free implementation of the DWG file format";
```

No need to duplicate the project name here:

```suggestion
    description = "Free implementation of the DWG file format";
```